src/Bicep.Core.IntegrationTests/RegistryProviderTests.cs

```diff
@@ -58,12 +59,71 @@ public async Task Providers_published_to_a_registry_can_be_compiled()
+        var fileSystem = FileHelper.CreateMockFileSystemForEmbeddedFiles(
+            typeof(RegistryProviderTests).Assembly,
+            "Files/RegistryProviderTests/HttpProvider");
+
+        var registry = "example.azurecr.io";
+        var repository = $"test/provider/http";
+
+        var services = GetServiceBuilder(fileSystem, registry, repository);
+
+        await DataSetsExtensions.PublishProviderToRegistryAsync(services.Build(), "/types/index.json", $"br:{registry}/{repository}:1.2.3");
+
+        var result = await CompilationHelper.RestoreAndCompile(services, """
+provider 'br:example.azurecr.io/test/provider/http@1.2.3' with {{}}
+
+resource dadJoke 'request@v1' = {
+  uri: 'https://icanhazdadjoke.com'
+  method: 'GET'
+  format: 'json'
+}
+
+output joke string = dadJoke.body.joke
+""");
 
-        // TODO remove this once the above is uncommented
-        result.Should().ContainDiagnostic("BCP204", DiagnosticLevel.Error, "Provider namespace \"http\" is not recognized.");
         result.Should().NotGenerateATemplate();
+        result.ExcludingLinterDiagnostics().Should().HaveDiagnostics(new[] {
+            //Harsh - Why are there multiple errors showing? Is this okay? 
+            ("BCP205", DiagnosticLevel.Error, "Provider namespace \"http\" does not support configuration."),
+            ("BCP022", DiagnosticLevel.Error, "Expected a property name at this location."),
+            ("BCP018", DiagnosticLevel.Error, "Expected the \":\" character at this location."),
+            ("BCP012", DiagnosticLevel.Error, "Expected the \"as\" keyword at this location.")
+        });
+    }
+
+    [TestMethod]
+    public async Task Third_party_imports_are_disabled_unless_feature_is_enabled()
+    {
+        var service = new ServiceBuilder();
+        var result = await CompilationHelper.RestoreAndCompile(service, @$"
+provider 'br:example.azurecr.io/test/provider/http@1.2.3'
+");
+        result.Should().HaveDiagnostics(new[] {
+                ("BCP203", DiagnosticLevel.Error, "Using provider statements requires enabling EXPERIMENTAL feature \"Extensibility\"."),
+            });
+
+        service = new ServiceBuilder().WithFeatureOverrides(new(ExtensibilityEnabled: true));
+        var result2 = await CompilationHelper.RestoreAndCompile(service, @$"
+provider 'br:example.azurecr.io/test/provider/http@1.2.3'
+");
+        result2.Should().HaveDiagnostics(new[] {
+                ("BCP204", DiagnosticLevel.Error, "Provider namespace \"http\" is not recognized."),
+            });
+
+        service = new ServiceBuilder().WithFeatureOverrides(new(ExtensibilityEnabled: true, ProviderRegistry:true));
+        var result3 = await CompilationHelper.RestoreAndCompile(service, @$"
+provider 'br:example.azurecr.io/test/provider/http@1.2.3'
+");
+        result3.Should().NotHaveAnyDiagnostics();
+        result3.Template.Should().NotBeNull();
```

Comment:
It's important to assert on the overall template that is generated - to make sure that it's well formed and that the "provider" metadata (name & version) have been correctly output.

See here for an example: https://github.com/Azure/bicep/blob/81e75fc4b474a0c00c64735a839c57cb5b12e0ad/src/Bicep.Core.IntegrationTests/ExtensibilityTests.cs#L510